src/shogun/multiclass/KNN.cpp

```diff
@@ -237,12 +242,17 @@ CMulticlassLabels* CKNN::classify_NN()
 
 SGMatrix<int32_t> CKNN::classify_for_multiple_k()
 {
-	ASSERT(m_num_classes>0)
-	ASSERT(distance)
-	ASSERT(distance->get_num_vec_rhs())
+	REQUIRE(distance, "No distance assigned.\n");
```

Comment:
use a consistent style for things not being set
"no X provided"
"X not set"
or something like this. But make it the same everywhere